File: app/src/main/java/com/shrikantbadwaik/tmdb/data/DataSource.java
package com.shrikantbadwaik.tmdb.data;

import com.shrikantbadwaik.tmdb.data.remote.UpcomingMoviesApi;
import com.shrikantbadwaik.tmdb.data.remote.apiresponse.ImageResponse;
import com.shrikantbadwaik.tmdb.data.remote.apiresponse.MovieResponse;
import com.shrikantbadwaik.tmdb.data.repository.Repository;
import com.shrikantbadwaik.tmdb.domain.helper.Constants;

import javax.inject.Inject;
import javax.inject.Singleton;

import retrofit2.Call;

@Singleton
public class DataSource implements Repository {
    private final UpcomingMoviesApi upcomingMoviesApi;

    @Inject
    public DataSource(UpcomingMoviesApi upcomingMoviesApi) {
        this.upcomingMoviesApi = upcomingMoviesApi;
    }

    @Override
    public Call<MovieResponse> upcomingMovies() {
        return upcomingMoviesApi.upcomingMovies(Constants.API_KEY);
    }

    @Override
    public Call<ImageResponse> moviePoster(String movieId) {
        return upcomingMoviesApi.moviePosters(movieId, Constants.API_KEY);
    }
}

File: app/src/main/java/com/shrikantbadwaik/tmdb/data/DataSource.kt
package com.shrikantbadwaik.tmdb.data

import com.shrikantbadwaik.tmdb.data.remote.TMDbApi
import com.shrikantbadwaik.tmdb.data.remote.apiresponse.MovieResponse
import com.shrikantbadwaik.tmdb.data.repository.Repository
import com.shrikantbadwaik.tmdb.domain.Constants
import io.reactivex.Observable
import java.util.*
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class DataSource @Inject constructor(
    private val tmDbApi: TMDbApi
) : Repository {

    override fun getUpcomingMovies(): Observable<MovieResponse> {
        return tmDbApi.upcomingMovies(Constants.API_KEY, Locale.getDefault().language)
    }
}